### DISCUSSION TITLE
Can I update outdated dependencies one by one?

### DISCUSSION THREAD

**Discussion by Dgame** [rectorphp/rector#8490] [Q&A] [answered]
In our case, we have a PHP 7.1 code base with a Guzzle 5.x version. We want to upgrade to a less outdated PHP Version (babysteps) soon but of course need to upgrade Guzzle at least to 6.x. Is that possible?

**Comment by samsonasik** [comment, marked as answer]
Upgrading dependencies require multiple steps, and yes, you need to do one by one, you can take a look how I upgrade goaop/parser-reflection dependencies along with code compatiblity itself.

https://github.com/goaop/parser-reflection/pulls?q=is%3Apr+author%3Asamsonasik+is%3Aclosed+sort%3Acreated-asc
